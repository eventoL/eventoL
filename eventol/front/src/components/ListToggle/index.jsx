import React from 'react';

import './index.scss';


export default class ListToggle extends React.PureComponent {
  state = {toggled: false};

  handleClick(){
    const {toggled} = this.state;
    this.setState({toggled: !toggled});
  }

  render(){
    const {toggled} = this.state;
    return (
      <div
        className='list-toggle'
        data-toggled={toggled}
        onKeyPress={this.handleClick}
        onClick={this.handleClick}
        role='menuitem'
        tabIndex='0'
      >
        <div>
          <i className='fa fa-fw fa-plus' />
          <i className='fa fa-fw fa-check' />
        </div>
      </div>
    );
  }
}
