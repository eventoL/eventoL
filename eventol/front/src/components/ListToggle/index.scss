@import '../../styles/variables';
@import '../../styles/commons';

.list-toggle {
  border: 2px solid $black;
  border-radius: 32px;
  font-size: 16px;
  height: 32px;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  right: 20px;
  top: 20px;
  width: 32px;

  &[data-toggled='true'] {
    background: $eventol-color;
    border: 2px solid $eventol-color;

    div {
      top: -32px;
    }
  }

  div {
    -webkit-transition: .125s ease;
    height: 64px;
    left: 0;
    position: absolute;
    top: 0;
    transition: .125s ease;
    width: 32px;

    i {
      @extend %flex-element;
      -ms-flex-align: center;
      -ms-flex-pack: center;
      -webkit-align-items: center;
      -webkit-box-align: center;
      -webkit-box-pack: center;
      -webkit-justify-content: center;
      align-items: center;
      height: 32px;
      justify-content: center;
      width: 32px;
    }
  }
}

.list-toggle:hover {
  border: 2px solid $black;

  div {
    top: -32px;
  }
}
