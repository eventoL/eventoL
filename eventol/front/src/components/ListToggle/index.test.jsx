import wait from 'waait';
import React from 'react';
import renderer from 'react-test-renderer';

import ListToggle from '.';

describe('ListToggle', () => {
  let tree;
  let element;
  let instance;
  let component;

  const getComponent = () => {
    element = <ListToggle />;
    component = renderer.create(element);
    instance = component.root;
    return component;
  };

  beforeEach(() => {
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });

    test('Snapshot with toggle change (onClick)', async () => {
      expect(tree).toMatchSnapshot();
      expect(component.getInstance().state.toggled).toBeFalsy();
      expect(
        instance.findByProps({className: 'list-toggle'}).props['data-toggled']
      ).toBeFalsy();

      instance.findByProps({className: 'list-toggle'}).props.onClick();
      component.update(element);
      await wait(0);

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      expect(component.getInstance().state.toggled).toBeTruthy();
      expect(
        instance.findByProps({className: 'list-toggle'}).props['data-toggled']
      ).toBeTruthy();
    });
  });
});
