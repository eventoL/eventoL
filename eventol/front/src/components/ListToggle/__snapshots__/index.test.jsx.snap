// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListToggle Default Snapshot 1`] = `
<div
  className="list-toggle"
  data-toggled={false}
  onClick={[Function]}
  onKeyPress={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <div>
    <i
      className="fa fa-fw fa-plus"
    />
    <i
      className="fa fa-fw fa-check"
    />
  </div>
</div>
`;

exports[`ListToggle Default Snapshot with toggle change (onClick) 1`] = `
<div
  className="list-toggle"
  data-toggled={false}
  onClick={[Function]}
  onKeyPress={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <div>
    <i
      className="fa fa-fw fa-plus"
    />
    <i
      className="fa fa-fw fa-check"
    />
  </div>
</div>
`;

exports[`ListToggle Default Snapshot with toggle change (onClick) 2`] = `
<div
  className="list-toggle"
  data-toggled={true}
  onClick={[Function]}
  onKeyPress={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <div>
    <i
      className="fa fa-fw fa-plus"
    />
    <i
      className="fa fa-fw fa-check"
    />
  </div>
</div>
`;
